imports/plugins/core/discounts/client/components/form.js

```diff
@@ -9,7 +9,7 @@ export default class DiscountForm extends Component {
     this.state = {
       discount: this.props.discount,
       validationMessage: null,
-      validatedInput: false,
+      validatedInput: false || this.props.validatedInput,
```

Comment:
should this be reversed as the `false` will always execute the second part of that statement.

```
this.props.validatedInput || false
```